nova/nova.go

```diff
@@ -590,3 +590,38 @@ func (c *Client) SetServerMetadata(serverId string, metadata map[string]string)
 	}
 	return err
 }
+
+// PortFixedIP represents a FixedIP with ip addresses and an associated
+// subnet id.
+type PortFixedIP struct {
+	IPAddress string `json:"ip_address"`
+	SubnetID  string `json:"subnet_id"`
+}
+
+// OSInterface represents an interface attachment to a server.
+type OSInterface struct {
+	FixedIPs   []PortFixedIP `json:"fixed_ips,omitempty"`
+	IPAddress  string        `json:"ip_address"`
+	MacAddress string        `json:"mac_addr,omitempty"`
+	NetID      string        `json:"net_id,omitempty"`
+	PortID     string        `json:"port_id,omitempty"`
+	PortState  string        `json:"port_state,omitempty"`
+}
```

Comment:
Just for completeness.

```suggestion
	Tag         string        `json:"tag,omitempty"`
}
```